tests/workorder_json.py

```diff
@@ -0,0 +1,226 @@
+# Copyright 2019 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import random
+import sys
+from utils.tcf_types import WorkerType
+
+class WorkOrderJson:
+	def __init__(self, _id):
+		self.json_obj = json.loads('{"jsonrpc":"2.0","method":"","id":"","params":{ } }')
+		self.set_id(_id)
+
+	def set_method(self, method):
+		self.json_obj["method"] = method
+
+	def get_id(self):
+		return self.json_obj["id"]
+
+	def set_id(self, _id):
+		self.json_obj["id"] = _id
+
+	def get_json_str(self):
+		return json.dumps(self.json_obj)
+
+	def get_json_str_indent(self):
+		return json.dumps(self.json_obj, indent=4)
+
+	def load_from_str(self, json_str):
+		self.json_obj = json.loads(json_str)
+
+class WorkerLookupJson(WorkOrderJson):
```

Comment:
Why is WorkerLookupJson inheriting from WorkOrderJson ? Every WorkOrder request contains worker_id. You might want to choose between is-a (inheritance) vs has-a(association) class relationship.

Or you can keep these class independent. Worker is different from work order.  We first create a worker. Then you can use the workerId and workloadId to create work order Id

Reply:
I think there is confusion because I named the WorkOrderJson class poorly, I have moved it to JsonRpcRequest and added a new WorkOrderJson class from which the WorkOrder requests inherit.